Libplanet/Blockchain/BlockChain.Validate.cs

```diff
@@ -159,28 +241,16 @@ internal InvalidBlockException ValidateNextBlockHeader(Block<T> block)
                 string message =
                     "The protocol version is disallowed to be downgraded from the topmost block " +
                     $"in the chain ({actualProtocolVersion} < {Tip.ProtocolVersion}).";
-                return new InvalidBlockProtocolVersionException(message, actualProtocolVersion);
-            }
-
-            if (Policy.ValidateNextBlock(this, block) is { } bpve)
-            {
-                return bpve;
+                throw new InvalidBlockProtocolVersionException(message, actualProtocolVersion);
             }
 
             Block<T> lastBlock = this[index - 1];
-            BlockHash? prevHash = lastBlock?.Hash;
-            DateTimeOffset? prevTimestamp = lastBlock?.Timestamp;
-
-            if (block.Index != index)
-            {
-                return new InvalidBlockIndexException(
-                    $"The expected index of block {block.Hash} is #{index}, " +
-                    $"but its index is #{block.Index}.");
-            }
+            BlockHash prevHash = lastBlock.Hash;
+            DateTimeOffset prevTimestamp = lastBlock.Timestamp;
```

Comment:
`nullable` is removed because the genesis block will not pass through, Am I correct? 🤔

Reply:
Yup